src/compiler/transformers/classFields.ts

```diff
@@ -266,8 +507,11 @@ namespace ts {
 
         function transformConstructor(node: ClassDeclaration | ClassExpression, isDerivedClass: boolean) {
             const constructor = visitNode(getFirstConstructorWithBody(node), visitor, isConstructorDeclaration);
-            const containsPropertyInitializer = forEach(node.members, isInitializedProperty);
-            if (!containsPropertyInitializer) {
+            const containsPropertyInitializerOrPrivateIdentifier = forEach(
+                node.members,
+                member => isInitializedProperty(member) || isPrivateIdentifierPropertyDeclaration(member)

```

Comment:
I would extract this to a function declaration outside of `transformConstructor` to avoid the possible overhead of allocating a new function object each time `transformConstructor` is called.

Reply:
Is this hot path? I'd be more concerned about the `forEach`s in the parser and checker